meta-mender-tn-imx-bsp/recipes-bsp/u-boot/u-boot-tn-imx/edm-g-imx8mp/0001-configs-edm-g-imx8mp-add-Mender-support.patch

```diff
@@ -24,10 +24,11 @@ index 5d55972019..c736004c34 100644
 +CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 +CONFIG_BOOTCOUNT_LIMIT=y
 +CONFIG_BOOTCOUNT_ENV=y
++CONFIG_BOOTCOUNT_BOOTLIMIT=3
```

Comment:
Why `CONFIG_BOOTCOUNT_BOOTLIMIT=3`? This not something that is typically needed. I think the default is 1?

Reply:
I tested with images imx-* images. This images has a service that is trying to resize the root partition and reboots afterwards.
U-boot counted this reboot as fail and booted old firmware.
With bootlimit = 3 I was able to successfully upgrade firmware with mender.
I belive bootlimit =2 would work too but I noticed that for other boards bootlimit is set to 3.

To get imx-* images fully supported by mender, the resizing script should be remove, which is not part of this repository.

Reply:
Very good catch @mirzak!

But this is quite specific and, for a user, unexpected behaviour then. This certainly mandates a board specific note in the Readme then.

Reply:
> I tested with images imx-* images. This images has a service that is trying to resize the root partition and reboots afterwards.

I think disabling these scripts would the correct approach. Is it not possible with a bbappend here?

I think setting `bootlimit=3` will break rollback functionality of the Mender client as it expects the bootloader to perform a rollback if it reboots due to failure in e.g committing. @TheYoctoJester , maybe you can verify with someone from the client team.

Reply:
Yes, bootlimit has to be 1, otherwise the client may misbehave.

Reply:
Thank you all for the feedback and the catch!

I will come back once I would be able to boot and upgrade without `bootlimit=3`.

Reply:
Thanks! I also think that disabling such a script via an append should not hard to add.